homeassistant/components/heos/config_flow.py

```diff
@@ -20,15 +20,22 @@ class HeosFlowHandler(config_entries.ConfigFlow):
 
     VERSION = 1
     CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+    DISCOVERED_HOSTS = {}
 
     async def async_step_discovery(self, discovery_info):
         """Handle a discovered Heos device."""
+        friendly_name = "{} ({})".format(
+            discovery_info[CONF_NAME], discovery_info[CONF_HOST])
+        self.DISCOVERED_HOSTS[friendly_name] = discovery_info[CONF_HOST]
+        # Only a single entry is needed for all devices
+        entries = self.hass.config_entries.async_entries(DOMAIN)
+        if entries:
+            return self.async_abort(reason='already_setup')
         # Only continue if this is the only active flow
         flows = self.hass.config_entries.flow.async_progress()
```

Comment:
You can call `self._async_in_progress()` to get all flows that are in progress for this handler that are not the one you're calling it from.